kernel/comps/framebuffer/src/pixel.rs

```diff
@@ -0,0 +1,106 @@
+    Grayscale8,
+    /// Each pixel uses 16 bits, with 5 bits for Red, 6 bits for Green, and 5 bits for Blue.
+    Rgb565,
+    /// Each pixel uses 24 bits, with 8 bits for Red, 8 bits for Green, and 8 bits for Blue.
+    Rgb888,
+    /// Each pixel uses 32 bits, with 8 bits each for Red, Green, Blue, and Alpha (transparency).
+    Rgba,
+}
+
+/// A rendered pixel in a specific format.
+#[derive(Debug, Default, Copy, Clone)]
+pub struct RenderedPixel {
+    buf: [u8; 4],
+    len: u8,
+}
+
+impl Pixel {
+    /// Renders the pixel into a specific format.
+    pub fn render(&self, format: PixelFormat) -> RenderedPixel {
+        let mut rendered_pixel = RenderedPixel::default();
+        match format {
+            PixelFormat::Grayscale8 => {
+                // Calculate the grayscale value
+                let red_weight = 77 * self.red as u16; // Equivalent to 0.299 * 256
+                let green_weight = 150 * self.green as u16; // Equivalent to 0.587 * 256
+                let blue_weight = 29 * self.blue as u16; // Equivalent to 0.114 * 256
+                let grayscale = (red_weight + green_weight + blue_weight) >> 8; // Normalize to 0-255
+                rendered_pixel.buf[0] = grayscale as u8;
+            }
+            PixelFormat::Rgb565 => {
+                let r = (self.red >> 3) as u16; // Red (5 bits)
+                let g = (self.green >> 2) as u16; // Green (6 bits)
+                let b = (self.blue >> 3) as u16; // Blue (5 bits)
+                let rgb565 = (r << 11) | (g << 5) | b; // Combine into RGB565 format
+                rendered_pixel.buf[0..2].copy_from_slice(&rgb565.to_be_bytes());
+            }
+            PixelFormat::Rgb888 => {
+                rendered_pixel.buf[0] = self.red;
+                rendered_pixel.buf[1] = self.green;
+                rendered_pixel.buf[2] = self.blue;
+            }
+            PixelFormat::Rgba => {
+                rendered_pixel.buf[0] = self.red;
+                rendered_pixel.buf[1] = self.green;
+                rendered_pixel.buf[2] = self.blue;
+                rendered_pixel.buf[3] = self.alpha;
+            }
+        }
+        rendered_pixel.len = format.nbytes() as u8;
+        rendered_pixel
+    }
+}
+
+impl PixelFormat {
+    /// Returns the number of bytes per pixel (color depth).
+    pub fn nbytes(&self) -> usize {
+        match self {
+            PixelFormat::Grayscale8 => 1,
+            PixelFormat::Rgb565 => 2,
+            PixelFormat::Rgb888 => 3,
+            PixelFormat::Rgba => 4,
+        }
+    }
+}
+
+impl RenderedPixel {
+    /// Returns the number of bytes in the rendered pixel.
+    pub fn nbytes(&self) -> usize {
+        self.len as usize
+    }
+
+    /// Returns a slice to the rendered pixel data.
+    pub fn as_slice(&self) -> &[u8] {
+        &self.buf[..self.nbytes()]
+    }
+}
+
+impl Pixel {
+    pub const WHITE: Pixel = Pixel {
+        red: 0xFF,
+        green: 0xFF,
+        blue: 0xFF,
+        alpha: 0xFF,
```

Comment:
Looks like we should not have the alpha field, otherwise the WHITE pixel will become transparent.

Framebuffer cannot support the alpha channel after all. The RGBA format is actually `PixelBlueGreenRedReserved8BitPerColor` ([UEFI spec](https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html?highlight=pixelformat#efi-graphics-output-protocol)).

So should we change `PixelFormat::Rgba` to `PixelFormat::RgbReserved`? Or maybe we can think of a better name, but I don't have a better idea at the moment.